src/notebooksyncagent.cpp

```diff
@@ -804,31 +836,52 @@ bool NotebookSyncAgent::calculateDelta(
         LOG_CRITICAL("mKCal::ExtendedStorage::deletedIncidences() failed");
         return false;
     }
-    Q_FOREACH (KCalCore::Incidence::Ptr incidence, deleted) {
-        bool uriWasEmpty = false;
-        QString remoteUri = incidenceHrefUri(incidence, mRemoteCalendarPath, &uriWasEmpty);
-        if (remoteUriEtags.contains(remoteUri)) {
-            if (uriWasEmpty) {
-                // we originally upsynced this pure-local addition, but then connectivity was
-                // lost before we updated the uid of it locally to include the remote uri.
-                // subsequently, the user deleted the incidence.
-                // Hence, it exists remotely, and has been deleted locally.
-                LOG_DEBUG("have local deletion for partially synced incidence:" << incidence->uid() << incidence->recurrenceId().toString());
+    // i = 1, deal with deletions of entire series / non-recurrence-id incidences
+    // i = 2, deal with deletions of recurrence-id incidences (persistent exception occurrences)
+    QSet<QString> deletedSeriesUids;
+    for (int i = 1; i <= 2; ++i) {
```

Comment:
Fair enough.  I can macroise it if that'd be better.  Splitting it into a function seems suboptimal as it's not really any more "special" than any other codeblock in this function, but I can see why the loop is confusing.